site/app/libraries/database/DatabaseQueries.php

```diff
@@ -7471,6 +7471,24 @@ public function getUserGroups(string $user_id): array {
         return $this->submitty_db->rows();
     }
 
+    public function getOtherCoursesWithSameGroup(string $semester, string $course): array {
+        $this->submitty_db->query(
+            'SELECT group_name FROM courses WHERE semester = ? AND course = ?',
+            [$semester, $course]
+        );
+
+        if (count($this->submitty_db->rows()) > 0) {
+            $this->submitty_db->query(
+                'SELECT course, semester FROM courses WHERE group_name = ?',
+                [$this->submitty_db->rows()[0]['group_name']]
+            );
+            return $this->submitty_db->rows();
+        }
+        else {
+            return [];
+        }
+    }
```

Comment:
This could be done in 1 query: ```SELECT c2.course, c2.semester FROM courses c1 INNER JOIN courses c2 ON c1.group_name = c2.group_name WHERE c1.semester = ? AND c1.course = ?```